cumulusci/tasks/bulkdata.py

```diff
@@ -485,7 +487,7 @@ def produce_csv():
                     sf_id = row[0]
                     yield "{},{}\n".format(local_id, sf_id).encode("utf-8")
                 else:
-                    self.logger.warning("      Error on row {}: {}".format(i, row[3]))
+                    raise BulkDataException("Error on row {}: {}".format(i, row[3]))
```

Comment:
I'm concerned that this may be too sensitive. If a handful of records fail due to lock timeouts in the middle of an 8-hour LDV bulk load, do we really want to have to start over?

Reply:
On the other hand, on non-volume-oriented loads, lack of some specific record invalidates the state of the org.

How about having fail-on-row-error be the default with a switch to allow row-level errors to be silently ignored?

Reply:
(I also have some improvements to the tests yet to push; I realized one of them was missing a bit of coverage.)

Reply:
Sure, an option to ignore row-level errors sounds like the right escape valve for that scenario.